pkg/prometheus/common_test.go

```diff
@@ -159,3 +160,125 @@ func TestBuildCommonPrometheusArgsWithRemoteWriteMessageV2(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildCommonPrometheusArgsWithOTLPReceiver(t *testing.T) {
+	for _, tc := range []struct {
+		version                    string
+		enableOTLPReceiver         *bool
+		expectedOTLPReceiverFlag   bool
+		OTLPConfig                 *monitoringv1.OTLPConfig
+		expectedOTLPFeatureEnabled bool
+	}{
+		// OTLP receiver not supported.
+		{
+			version:                    "2.46.0",
+			enableOTLPReceiver:         ptr.To(true),
+			expectedOTLPFeatureEnabled: false,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver supported starting with v2.47.0.
+		{
+			version:                    "2.47.0",
+			enableOTLPReceiver:         ptr.To(true),
+			expectedOTLPFeatureEnabled: true,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver supported but not enabled.
+		{
+			version:                    "2.47.0",
+			enableOTLPReceiver:         ptr.To(false),
+			expectedOTLPFeatureEnabled: false,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver config supported but verison not support
+		{
+			version:            "2.46.0",
+			enableOTLPReceiver: ptr.To(false),
+			OTLPConfig: &monitoringv1.OTLPConfig{
+				PromoteResourceAttributes: []string{"aa", "bb"},
+			},
+			expectedOTLPFeatureEnabled: false,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver config supported
+		{
+			version:            "2.55.0",
+			enableOTLPReceiver: nil,
+			OTLPConfig: &monitoringv1.OTLPConfig{
+				PromoteResourceAttributes: []string{"aa", "bb"},
+			},
+			expectedOTLPFeatureEnabled: true,
+			expectedOTLPReceiverFlag:   false,
+		},
+		// OTLP receiver config supported with verison 3.x
```

Comment:
```suggestion
		// OTLP receiver config supported with version 3.x
```

Reply:
Since this changes comes from upstream, need to fix typo there